src/main/java/com/sngular/kloadgen/extractor/extractors/JsonExtractor.java

```diff
@@ -19,18 +18,40 @@
 import com.sngular.kloadgen.model.json.ObjectField;
 import com.sngular.kloadgen.model.json.Schema;
 import com.sngular.kloadgen.model.json.StringField;
+import io.confluent.kafka.schemaregistry.ParsedSchema;
+import io.confluent.kafka.schemaregistry.json.JsonSchema;
 import org.apache.commons.collections4.CollectionUtils;
 import org.apache.commons.collections4.Transformer;
 import org.apache.commons.lang3.StringUtils;
 
-public class JsonExtractor {
+public class JsonExtractor implements Extractor{
+
+  public static final String ARRAY = "-array";
 
   private final JSONSchemaParser jsonSchemaParser = new JSONSchemaParser();
 
-  public final List<FieldValueMapping> processSchema(final JsonNode jsonNode) {
+  public final List<FieldValueMapping> processSchema(final Object schema) {
+    var jsonNode=((JsonSchema) schema).toJsonNode();
     return processSchema(jsonSchemaParser.parse(jsonNode));
   }
 
+  public final ParsedSchema getParsedSchema(final String schema){
+    return new JsonSchema(schema);
+  }
+
+  public List<FieldValueMapping> processApicurioParsedSchema(final Object schema){
```

Comment:
A side effect of this solution is that we have to modify all schema types if a new schema registry provider is included.